src/OpenTelemetry/Trace/BatchExportActivityProcessor.cs

```diff
@@ -127,6 +133,82 @@ public override void OnEnd(Activity activity)
             Interlocked.Increment(ref this.droppedCount);
         }
 
+        /// <summary>
+        /// Flushes the <see cref="Activity"/> currently in the queue, blocks
+        /// the current thread until flush completed, shutdown signaled or
+        /// timed out. Using a 32bit signed  integer to specify the time
+        /// interval in milliseconds.
+        /// </summary>
+        /// <param name="timeoutMillis">
+        /// The number of milliseconds to wait, or <c>Timeout.Infinite</c> to
+        /// wait indefinitely.
+        /// </param>
+        /// <returns>
+        /// Returns <c>true</c> when flush completed; otherwise, <c>false</c>.
+        /// </returns>
+        public bool ForceFlush(int timeoutMillis = Timeout.Infinite)
+        {
+            if (timeoutMillis < 0 && timeoutMillis != Timeout.Infinite)
+            {
+                throw new ArgumentOutOfRangeException(nameof(timeoutMillis));
+            }
+
+            var tail = this.queue.RemovedCount;
+            var head = this.queue.AddedCount;
+
+            if (head == tail)
+            {
+                return true; // nothing to flush
+            }
+
+            this.exportTrigger.Set();
+
+            if (timeoutMillis == 0)
+            {
+                return false;
+            }
+
+            var triggers = new[] { this.dataExportedNotification, this.shutdownTrigger };
+
+            var sw = Stopwatch.StartNew();
+
+            try
+            {
+                while (true)
+                {
+                    if (timeoutMillis == Timeout.Infinite)
+                    {
+                        WaitHandle.WaitAny(triggers);
+                    }
+                    else
+                    {
+                        var timeout = (long)timeoutMillis - sw.ElapsedMilliseconds;
+
+                        if (timeout <= 0)
+                        {
+                            return this.queue.RemovedCount >= head;
+                        }
+
+                        WaitHandle.WaitAny(triggers, (int)timeout);
+                    }
+
+                    if (this.queue.RemovedCount >= head)
+                    {
+                        return true;
+                    }
+
+                    if (this.shutdownTrigger.WaitOne(0))
+                    {
+                        return false;
+                    }
+                }
+            }
+            finally
+            {
+                sw.Stop();
```

Comment:
You don't need to call `Stop()` for a stopwatch that you are not going to use anymore. You can get rid of the try-finally.